### PR TITLE
Implement retry feature

### PR DESCRIPTION
**Pull request by geobeau**
See README change for description of change

### CODE REVIEW

**Review by Lqp1 — APPROVED** [review]
👍🏻

**Comment by josqu4red** [comment]
Rebase needed

File: README.md [hunk 1]
```diff
@@ -147,7 +154,7 @@ running inside Kubernetes. It can select Pod (like PDB) but also PVC (to protect
 ApplicationDisruptionBudget aims at providing a way to expose service level healthiness to Kubernetes.
 In some cases, an application can be unhealthy even if all its pods are running.
 
-You can select Pods and/or PVCs.
+You can select Pods and/or PVCs.
```
Comment (Lqp1) [inline README.md:157]: :D ?

**Review by Lqp1 — APPROVED** [review]

File: internal/controller/nodedisruption_controller_test.go [hunk 2]
```diff
@@ -36,6 +39,30 @@ func clearAllNodeDisruptionRessources() {
 	_ = k8sClient.DeleteAllOf(context.Background(), &nodedisruptionv1alpha1.ApplicationDisruptionBudget{})
 }
 
+func start_reconciler_with_config(config NodeDisruptionReconcilerConfig) (cancel_fn func()) {
```
Comment (josqu4red) [inline internal/controller/nodedisruption_controller_test.go:42]: cancel_fn context.CancelFunc ?

File: internal/controller/nodedisruption_controller_test.go [hunk 3]
```diff
@@ -219,6 +358,25 @@ var _ = Describe("NodeDisruption controller", func() {
 					}, timeout, interval).Should(Equal(nodedisruptionv1alpha1.Granted))
 				})
 			})
+
+			When("RejectEmptyNodeDisruption is enabled", func() {
+				It("grants the NodeDisruption", func() {
```
Comment (josqu4red) [inline internal/controller/nodedisruption_controller_test.go:363]: rejects

File: internal/controller/nodedisruption_controller_test.go [hunk 4]
```diff
@@ -85,89 +114,203 @@ var _ = Describe("NodeDisruption controller", func() {
...
+				clearAllNodeDisruptionRessources()
+			})
 
-				By("By checking the NodeDisruption is being granted")
-				Eventually(func() nodedisruptionv1alpha1.NodeDisruptionState {
-					err := k8sClient.Get(ctx, NDLookupKey, createdDisruption)
-					if err != nil {
-						panic("should be able to get")
+			Context("When there are no budgets in the cluster", func() {
+				It("It grants the node disruption", func() {
```
Comment (josqu4red) [inline internal/controller/nodedisruption_controller_test.go:134]: Note: no need to repeat the context word in text e.g `It("grants...` `By("checking...`

File: README.md (hunk 1 above)
Comment (geobeau) [reply README.md]: 🙊

**Pull request merged**